src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -544,14 +532,11 @@ public List<ClusterTransactionCommand.SingleClusterDatabaseCommand> ExecuteClust
                 //_nextClusterCommand refers to each individual put/delete while batch size refers to number of transaction (each contains multiple commands)
                 _logger.Info($"Read {batch.Count:#,#;;0} cluster transaction commands - fromCount: {_nextClusterCommand}, take: {batchSize}");
             }
-
+            
             if (batch.Count == 0)
             {
                 var index = _serverStore.Cluster.GetLastCompareExchangeIndexForDatabase(context, Name);
-
-                if (RachisLogIndexNotifications.LastModifiedIndex != index)
-                    RachisLogIndexNotifications.NotifyListenersAbout(index, e: null);
-
+                ClusterWideTransactionIndexWaiter.SetAndNotifyListenersIfHigher(index);
```

Comment:
We changed the implementation.
A cluster-wide transaction without a database command will not be written to the `ClusterStateMachine.TransactionCommands` table.
Instead, if we don't find any new commands we notify according to the last compare exchange index